The default of CONFIG_SYS_BOOTM_LEN is 8M which is not large enough when you
bundle the initramfs with the kernel image.

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 834f1cd236..80fbc50b21 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -59,6 +59,8 @@
 
 #ifdef CONFIG_ARM64
 #define CONFIG_SYS_BOOTM_LEN		SZ_64M
+#else
+#define CONFIG_SYS_BOOTM_LEN		SZ_16M
 #endif
 
 /* Devices */
